maven-model-builder/src/test/java/org/apache/maven/model/path/DefaultUrlNormalizerTest.java

```diff
@@ -82,8 +70,15 @@ public void testPreservationOfDoubleSlashes()
                       normalize( "[fetch=]http://server.org/[push=]ssh://server.org/" ) );
     }
 
-    public void testNormalizeInputWithNoParentDirectoriesToAscendTo()
+    @Test
+    public void relativeUriReferenceLeftUnaffectedWithNoParentDirectoryToAscendTo()
+    {
+        assertEquals( "/../", normalize("/../" ) );
```

Comment:
Did you test the same with `Path`?

Reply:
`Path` and `URI` differ slightly here; I've gone with [RFC 3986's Remove Dot Segments](https://tools.ietf.org/html/rfc3986#section-5.2.4) and adjusted the code to remove them.